### ISSUE TITLE
[feature] Add aria-label to wcm-modal component

### ISSUE THREAD

**Issue by cashd** [WalletConnect/modal#35]
### What problem does this new feature solve?

The [wcm-modal](https://github.com/WalletConnect/modal/blob/main/packages/modal-ui/src/partials/wcm-modal/index.ts) component does not pass accessibility standards for dialogs because it is missing an `aria-label`. I ran an accessibility test using Google Lighthouse and it raised this issue. 

Here is a reference to the accessibility standard https://dequeuniversity.com/rules/axe/4.9/aria-dialog-name

### Describe the solution you'd like

Add an `aria-label` property to the `wcm-modal` component to satisfy accessibility requirements for users with screen readers.
